environs/interface.go

```diff
@@ -232,6 +232,15 @@ type Environ interface {
 	// the Bootstrap method's job to create the controller model.
 	Create(CreateParams) error
 
+	// UpdateController is called when the model is moved from one controller
+	// to another using model migration. Some providers tag instances, disks,
+	// and cloud storage with the controller UUID to aid in clean destruction.
+	// All things that the provider has tagged with the controller UUID need
+	// to be updated in this method to use the new controller UUID. For
+	// providers that do not track the controller UUID, a simple method
+	// returning nil will suffice.
+	UpdateController(controllerUUID string) error
```

Comment:
Hmm... interesting idea.

We were planning to run this on the source controller as part of the work in the migration master.
Perhaps now we should do it as part of the import on the target controller. We do need to consider how to roll back...

I'll wait and chat to @mjs and @babbageclunk tomorrow.